### DISCUSSION TITLE
Merging Upstream Changes

### DISCUSSION THREAD

**Discussion by trentschnee** [Vagr9K/gatsby-material-starter#149] [Q&A]
Hi @Vagr9K . First of all, thanks for creating this repo. I've been trying to figure out how exactly I can clone gatsby-material-starter and I've just not been able to wrap my head around the Receiving upstream updates section in your README. 

My initial goal is to have a yarn workspaces so that I can make local changes to the theme without publishing and merge upstream changes but I'm uncertain on how to do that. Do I just need to clone this repo and replace the `examples/material-demo` workspace with my own custom workspace with the same package.json?  How would I merge updates to the gatsby-material-starter theme within git? I just don't understand that. Sorry for all the questions but I've looked up my questions but couldn't find any answers. Thanks for your time.

**Comment by Vagr9K** [comment]
Hey!

Don't focus on examples.

When you clone the repo, the root project is your project. It's already hooked up to themes for you to use.

The examples are more of a reference and are used to E2E test themes. You don't need to make changes to them.

Just clone the project, run `yarn` (workspaces are already setup out of the box) and then start modifying the root Gatsby project as you would do normally.

You'll also be able to make changes to `themes/` and those will get reflected in the root project.

**Reply by trentschnee** [reply]
Thanks for your response @Vagr9K that clarifies a lot of things up. However, I was wondering how I can pull/merge upstream changes in within my repo. I appreciate your time!

**Reply by Vagr9K** [reply]
_If you don't modify the themes_, you can just bump their versions in your project (instead of dealing with Git) since new theme versions get published on npm.

_Otherwise, if you do_, take a look at [this guide by GitHub](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/merging-an-upstream-repository-into-your-fork).

**Comment by trentschnee** [comment]
@Vagr9K thank you sir for your time and answering my questions!
